### PR TITLE
[WIP] eVaults existing work

### PR DESCRIPTION
**Pull request by rhiaro**
* Restructures section from headings in @dmitrizagidulin's doc
* Fills out new sections with content from the doc and what was there before and explanatory padding
* Adds a summary

TODO: 

* Figure out the table.

### CODE REVIEW

File: draft-documents/encrypted-data-vaults.md [hunk 1]
```diff
@@ -79,79 +69,172 @@ one or more device(s) they control, and data is stored locally to these devices.
...
+keys. There are roughly three approaches: storage-side encryption, client-side
+(edge) encryption, and gateway-side encryption (which is a hybrid of the
+previous two).
+
+Any data storage systems which let the user store arbitrary data support
+client-side encryption at the most basic level. That is, they let the user
+encrypt data themselves, and then store it. This doesn’t mean these systems are
+optimised for encrypted data however (eg. querying and sharing may be difficult),
+or that they won’t leak data in other ways (eg. through unencrypted metadata).
+Solid, NextCloud, Identity Hubs and IPFS take this approach.
```
Comment (dmitrizagidulin) [inline draft-documents/encrypted-data-vaults.md]: Maybe we can say something like:

This doesn’t mean these systems are optimised for encrypted data however. Querying and access control for encrypted data may be difficult (as is the case for Solid, NextCloud and IPFS). And querying un-encrypted metadata (as in the current design for Identity Hubs) may leak confidential information in other ways.

File: draft-documents/encrypted-data-vaults.md [hunk 2]
```diff
@@ -79,79 +69,172 @@ one or more device(s) they control, and data is stored locally to these devices.
...
+
+### Indexing and Querying
+
+Encrypted data which is opaque to the storage server introduces challenges for
+indexing and searching the contents of the data. Some systems work around this
+with a certain amount of unencrypted metadata attached to the data objects.
+Another possibility is unencrypted listings of pointers to filtered subsets
+of data.
+
+Resources in Solid are listed by URI in `ldp:Container`s, which serve as indexes
```
Comment (dmitrizagidulin) [inline draft-documents/encrypted-data-vaults.md]: I think referencing `ldp:Container` might be too detailed for the overview. Maybe something like:

Solid aims to provide a web accessible interface to a file system. Resources (linked data or arbitrary files) are organized into folder-like Containers. Precisely how the data is stored is an implementation detail ... [etc]

File: draft-documents/encrypted-data-vaults.md [hunk 3]
```diff
@@ -79,79 +69,172 @@ one or more device(s) they control, and data is stored locally to these devices.
...
+servers receive HTTP requests containing RDF payloads and create or modify the
+target URI accordingly.
+
+Data Hubs uses JSON Web Tokens (JWTs) and specified endpoints. Multiple requests
+are required, first to retrieve metadata for the desired data object(s), and then
+to retrieve the sequence of commits which make up the actual data. Mechanisms for
+authentication are still under development. Access control is carried out via
+posting to the Permissions interface.
+
+Tahoe-LAFS uses a client-server architecture, whereby a local client encrypts
```
Comment (dmitrizagidulin) [inline draft-documents/encrypted-data-vaults.md]: How do you feel about:

Tahoe-LAFS uses a client-gateway-storage server architecture, whereby a client passes data to a gateway server for encryption and chunking. The gateway, in turn, stores the individual chunks on a cluster of storage servers (several copies of the data are stored, for greater availability and disaster recovery). Services are identified ... [etc]

File: draft-documents/encrypted-data-vaults.md [hunk 4]
```diff
@@ -79,79 +69,172 @@ one or more device(s) they control, and data is stored locally to these devices.
...
+### Access Interface and Control
+
+Whether data is accessed over a network, or on a local device, data objects tend
+to need globally unique identifiers. Interfaces for reading and writing data in
+stores, as well as the mechanisms to restrict or grant the ability to do so, vary
+between implementations.
+
+NextCloud and Solid both make use of existing Web standards. NextCloud uses
+WebDAV to allow client applications to read, write, and search
+data on the server's filesystem using a directory structure, and OCP for authentication.
```
Comment (dmitrizagidulin) [inline draft-documents/encrypted-data-vaults.md]: Let's change "OCP for authentication" to:

"and a [custom login flow](https://docs.nextcloud.com/server/16/developer_manual/client_apis/LoginFlow/) for local authentication."

File: draft-documents/encrypted-data-vaults.md (hunk 1 above)
Comment (rhiaro) [reply draft-documents/encrypted-data-vaults.md]: Yes minus the last sentence because metadata and querying are covered more later

**Review by dmitrizagidulin — APPROVED** [review]

File: draft-documents/encrypted-data-vaults.md [hunk 5]
```diff
@@ -79,79 +69,175 @@ one or more device(s) they control, and data is stored locally to these devices.
...
+architecture have access to the (unencrypted) data, or who controls the private
+keys. There are roughly three approaches: storage-side encryption, client-side
+(edge) encryption, and gateway-side encryption (which is a hybrid of the
+previous two).
+
+Any data storage systems which let the user store arbitrary data support
+client-side encryption at the most basic level. That is, they let the user
+encrypt data themselves, and then store it. This doesn’t mean these systems are
+optimized for encrypted data however. Querying and access control for encrypted
+data may be difficult (as is the case for Solid, NextCloud, Data Hubs and IPFS).
```
Comment (msporny) [inline draft-documents/encrypted-data-vaults.md:89]: Hmm, "Data Hubs" use is prevalent here when I think what's meant is "Identity Hubs"? @dmitrizagidulin @rhiaro -- thoughts? I don't think the Secure Data Hubs spec does much of the stuff listed under "Data Hubs"?

Comment (dmitrizagidulin) [reply draft-documents/encrypted-data-vaults.md:89]: Yeah, typo on my part - i’ll fix

Comment (dmitrizagidulin) [reply draft-documents/encrypted-data-vaults.md:89]: @msporny Fixed in PR https://github.com/WebOfTrustInfo/rwot9-prague/pull/163

**Pull request merged**